components/constellation/constellation.rs

```diff
@@ -304,6 +312,97 @@ enum ExitPipelineMode {
     Force,
 }
 
+/// A logger directed at the constellation from content processes
+#[derive(Clone)]
+pub struct FromScriptLogger {
```

Comment:
We could probably have a `FromLogger<T>` struct with a common Log impl, where `T` is `FromFooMsg`, and implements a trait which provides a `get_msg(Option<PipelineId>, Option<String>, LogEntry) -> Self` method